src/app/content/highlights/components/SummaryPopup/FiltersList.tsx

```diff
@@ -0,0 +1,161 @@
+  archiveTreeSectionIsBook,
+  archiveTreeSectionIsChapter,
+  flattenArchiveTree,
+} from '../../../utils/archiveTreeUtils';
+import { setChaptersFilter, setColorsFilter } from '../../actions';
+import { filtersChange } from '../../actions';
+import { chaptersFilter, colorsFilter } from '../../selectors';
+
+// tslint:disable-next-line: variable-name
+const RemoveIcon = styled.span`
+  padding: 0.5rem;
+  cursor: pointer;
+
+  svg {
+    height: 1rem;
+    width: 1rem;
+  }
+`;
+
+// tslint:disable-next-line: variable-name
+const ItemLabel = styled.span`
+  max-width: 80px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  text-transform: capitalize;
+`;
+
+interface FiltersListColorItemProps {
+  type: 'colors';
+  color: HighlightColorEnum;
+  onRemoveItem: (type: 'colors', color: HighlightColorEnum) => void;
+}
+
+interface FiltersListChapterItemProps {
+  type: 'chapters';
+  title: string;
+  chapterId: string;
+  onRemoveItem: (type: 'chapters', chapterId: string) => void;
+}
+
+type FiltersListItemProps = FiltersListColorItemProps | FiltersListChapterItemProps;
+
+// tslint:disable-next-line: variable-name
+const FiltersListItem = (props: FiltersListItemProps) => {
+  const handleClick = () => {
+    if (props.type === 'colors') {
+      props.onRemoveItem(props.type, props.color);
+    } else {
+      props.onRemoveItem(props.type, props.chapterId);
+    }
+  };
+
+  let body: JSX.Element | string;
+  if (props.type === 'colors') {
+    body = <FormattedMessage id={`i18n:highlighting:colors:${props.color}`}>
+      {(msg: Element | string) => msg}
+    </FormattedMessage>;
+  } else {
+    body = <span dangerouslySetInnerHTML={{ __html: props.title }}/>;
+  }
+
+  return <li>
+    <RemoveIcon onClick={handleClick}><Times /></RemoveIcon>
+    <ItemLabel>{body}</ItemLabel>
+  </li>;
+};
+
+interface FiltersListProps {
+  className?: string;
+}
+
+type SectionsMap = Map<string, LinkedArchiveTree | LinkedArchiveTreeSection>;
+
+// tslint:disable-next-line: variable-name
+const FiltersList = ({className}: FiltersListProps) => {
+  const [sections, setSections] = React.useState<SectionsMap>(new Map());
+
+  const book = useSelector(bookSelector);
+  const selectedChapters = useSelector(chaptersFilter);
+  const selectedColors = useSelector(colorsFilter);
+
+  const dispatch = useDispatch();
+
+  const setSelectedChapters = (chapterIds: string[]) => {
+    dispatch(setChaptersFilter(chapterIds));
+    dispatch(filtersChange());
+  };
+  const setSelectedColors = (colors: HighlightColorEnum[]) => {
+    dispatch(setColorsFilter(colors));
+    dispatch(filtersChange());
+  };
+
+  const onRemoveItem = (type: 'colors' | 'chapters', label: HighlightColorEnum | string) => {
+    if (type === 'colors') {
+      setSelectedColors(selectedColors.filter(not(match(label))));
+    } else if (type === 'chapters') {
+      setSelectedChapters(selectedChapters.filter(not(match(label))));
+    }
+  };
+
+  React.useEffect(() => {
```

Comment:
i think you're doing this so that it doesn't reevaluate the whole tree anytime the component is re-rendered, it would be cleaner to use a selector to do this.